db/migrations/00031_notification_preferences.sql

```diff
@@ -0,0 +1,77 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE notification_types (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  name STRING NOT NULL,
+  slug STRING NOT NULL,
+  description STRING NOT NULL,
+  default_enabled BOOL NOT NULL,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  deleted_at TIMESTAMPTZ NULL,
+
+  CONSTRAINT notification_types_slug_key UNIQUE (slug) WHERE deleted_at IS NULL,
+  INDEX (deleted_at) STORING (slug, default_enabled)
+);
+
+CREATE TABLE notification_targets (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  name STRING NOT NULL,
+  slug STRING NOT NULL,
+  description STRING NOT NULL,
+  default_enabled BOOL NOT NULL,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  deleted_at TIMESTAMPTZ NULL,
+
+  CONSTRAINT notification_targets_slug_key UNIQUE (slug) WHERE deleted_at IS NULL,
+  INDEX (deleted_at) STORING (slug, default_enabled)
+);
+
+CREATE TABLE notification_preferences (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  user_id UUID NOT NULL REFERENCES users(id) ON DELETE CASCADE ON UPDATE CASCADE,
+  notification_type_id UUID NOT NULL REFERENCES notification_types(id) ON DELETE CASCADE ON UPDATE CASCADE, 
+  notification_target_id UUID NULL REFERENCES notification_targets(id) ON DELETE CASCADE ON UPDATE CASCADE,
+  notification_target_id_null_string UUID AS (IFNULL(notification_target_id, '00000000-0000-0000-0000-000000000000')) STORED,
+  enabled BOOL NOT NULL DEFAULT false,
+
+  CONSTRAINT unique_user_type_target UNIQUE (user_id, notification_type_id, notification_target_id_null_string),
+  INDEX (user_id) STORING (notification_type_id, notification_target_id, notification_target_id_null_string, enabled),
```

Comment:
there seems to be a lot of overlap here on these two indexes. 


`INDEX (user_id) STORING (notification_type_id, notification_target_id, notification_target_id_null_string, enabled) `

and 
`INDEX (user_id, notification_target_id, notification_type_id) STORING (enabled)`


The crdb query planner still has some interesting quirks to me, however in my head something like the following might work as a single index. 

` INDEX (user_id, notification_target_id, notification_type_id) STORING (enabled,  notification_target_id_null_string)`


However it is possible I'm overlooking something.

Reply:
I ran your index on the [query that created the suggestions for the first two indices](https://github.com/metal-toolbox/governor-api/pull/15/files#diff-4b14d273a00680e0d2e6e7c82e731bb0e45a73768b009676976b3a52b6d86e06R99) with `explain`, everything looks ok. I will commit the suggestion.